tiledb/sm/crypto/test/unit_tiledb_crypto.cc

```diff
@@ -493,50 +522,102 @@ TEST_CASE("Crypto: Test AES-256-GCM", "[crypto][aes]") {
   }
 }
 
+static std::string from_hex(const std::string_view& str) {
+  std::string output;
+  for (size_t i = 0; i < str.length(); i += 2) {
+    char byte_str[3] = {str[i], str[i + 1], '\0'};
+    output.push_back((uint8_t)std::strtoul(byte_str, nullptr, 16));
+  }
+  return output;
+}
+
+static std::string to_hex(span<const uint8_t> data) {
+  std::stringstream ss;
+  for (size_t i = 0; i < data.size(); i++) {
+    ss << std::hex << std::setw(2) << std::setfill('0') << (int)data[i];
+  }
+  return ss.str();
+}
+
+/**
+ * Test that the given input (optionally in hex) has the expected hash value in
+ * hex. The function is generic over the hash and the input size.
+ */
+template <Status Hash(const void*, uint64_t, Buffer*), int Digest_Bytes>
```

Comment:
Sizes should be `size_t` unless they need to be compatible with existing code that (incorrectly) uses any other type for in-memory sizes. Here's that's `Digest_Bytes`.

Nit: template arguments should be lower case.

Reply:
The functions in the `Crypto` class use `uint64_t`, but changing the test code here to `site_t` works. Updated.

Reply:
Reverted the change to `size_t` due to macOS CI failures. It will go away soon either way when we use spans instead of pointer-size pairs.